kayenta-newrelic-insights/src/main/java/com/netflix/kayenta/canary/providers/metrics/NewRelicCanaryMetricSetQueryConfig.java

```diff
@@ -24,22 +24,40 @@
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 import lombok.ToString;
+import org.springframework.util.StringUtils;
 
-@Builder
+@Builder(toBuilder = true)
 @ToString
 @NoArgsConstructor
 @AllArgsConstructor
 @JsonTypeName("newrelic")
 public class NewRelicCanaryMetricSetQueryConfig implements CanaryMetricSetQueryConfig {
 
+  public static final String SERVICE_TYPE = "newrelic";
+
   @Getter private String q;
 
   @NotNull @Getter private String select;
 
   @Getter private String resolution;
 
+  @Getter private String customInlineTemplate;
+
+  @Getter private String customFilterTemplate;
```

Comment:
I added this because this is how the system that @duftler made for templates works, but I am not 100 sure how filters work and if they are relevant to New Relic.

Reply:
@duftler is this specific to prometheus and or stack driver, or would this be relevant here too?

Reply:
`customFilterTemplate` refers by name to a template contained in the templates map on the canary config. `customInlinetemplate` directly contains the actual template to use. The prometheus canary config metric modal allows one to configure inline templates.

I think it's relevant for this provider as well.